### PR TITLE
feat(users): add filtering and sorting to users

### PR DESCRIPTION
**Pull request by ludtkemorgan**
# Pull Request Template

## Issue Overview

This PR addresses #1421 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Functionality copied from bloom core:
- Add filter to the Users page. This allows the filtering of Name, Email, and Listings. At least 3 characters must be typed for filtering to apply
image

New Functionality:
- Make all of the columns in the users table sortable. 

Other:
- Cleanup some random linting warnings/errors throughout the code

## How Can This Be Tested/Reviewed?

1. Go to the /users page and type into the filter box. Results should be filtered to contain only users that match your filter (either name, email, or listings contain the string)
2. Go to the /users page and sort on each column. Results should sort ascending and descending based on each column.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend/core/src/auth/dto/user-filter-type-to-field-map.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { UserFilterKeys } from "../types/user-filter-keys"
 
 export const userFilterTypeToFieldMap: Record<keyof typeof UserFilterKeys, string> = {
-  isPartner: "user_roles.isPartner",
-  isPortalUser: "user_roles",
+  isPartner: "userRoles.isPartner",
+  isPortalUser: "userRoles",
```
Comment (ludtkemorgan) [inline backend/core/src/auth/dto/user-filter-type-to-field-map.ts:5]: I'm not 100% sure why this change was needed. It was required to get my changes to work but is not in the original change in bloom core. If anyone knows why this could be or if I missed some change it would be much appreciated

Comment (seanmalbert) [reply backend/core/src/auth/dto/user-filter-type-to-field-map.ts:5]: It makes sense that it was `user_roles`, because the old `_getQb` function specified that as the alias. Looking at core/dev now and it does match what you have here, and it matches the join as set in the `getQb` function in `backend/core/src/auth/repositories/user-repository.ts`. So I think we're good.

**Review by seanmalbert — APPROVED** [review]

**Review by emilyjablonski — APPROVED** [review]
LGTM! Do you mind adding QA Notes to the issue before merge?
